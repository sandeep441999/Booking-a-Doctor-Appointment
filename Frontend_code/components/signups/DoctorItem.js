import Link from "next/link";
import { useRouter } from "next/router";
import { useState } from "react";

import Card from "../ui/Card";
import classes from "./DoctorItem.module.css";

const slots = [
  {
    time: "10-11 am",
    available: true,
  },
  {
    time: "11-12 am",
    available: true,
  },
  {
    time: "2-3 pm",
    available: true,
  },
  {
    time: "3-4 pm",
    available: true,
  },
];

//Rendering the doctoritem component to list each doctor item

function DoctorItem(props) {
  const [showSlots, setShowSlots] = useState(false);
  const router = useRouter();


  //below function is written whether to show timeslot for booking appointment or not.

  const showAppointmentsHandler = () => {
    console.log(showSlots);
    console.log(slots);
    setShowSlots((showSlots) => !showSlots);
  };

  const bookAppointmentHandler = () => {
    console.log(emailRef.current.value);
    router.push('/BookAppointment');
  };

  //Rendering each doctor details in a specific and attract style using the props received.

  return (
    <li className={classes.item}>
      <Card>
        <div className={classes.content} data-testid={`doctor-${props.id}`}>
          <h3>{props.Name}</h3>
          <address>
            <strong>Qualification: </strong>
            {props.Qualification}
          </address>
          <address>
            <strong>Speciality: </strong>
            {props.Speciality}
          </address>
          <address>
            <strong>Experience: </strong>
            {props.Experience}
          </address>
          <address>
            <strong>Email: </strong>
            {props.Email}
          </address>
        </div>
        <div className={classes.actions}>
          <button onClick={showAppointmentsHandler}>Book an Appointment</button>
        </div>
        <div className={classes.actions}>
          {/* {showSlots && slots.map((slot) => <button onClick={bookAppointmentHandler}>{slot.time}</button>)} */}
          {showSlots && slots.map((slot) => <button><Link href={{ pathname: '/BookAppointment', query: { email: props.Email, timeSlot: slot.time } }} >{slot.time}</Link></button>)}
        </div>
      </Card>
    </li>
  );
}

export default DoctorItem;
