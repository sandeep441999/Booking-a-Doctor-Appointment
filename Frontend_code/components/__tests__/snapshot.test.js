import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
//import "jest-dom/extend-expect";
import '@testing-library/jest-dom'

import DoctorItem from "../signups/DoctorItem";



test("matches snapshot", () => {
  const doctor = {
    id: "d1",
    Name: "Sam",
    Age: "45",
    Speciality: "Cardiology",
    Qualification: "MBBS",
  };
  const tree = renderer
    .create(
      <DoctorItem
        key={doctor.id}
        id={doctor.id}
        Name={doctor.Name}
        Qualification={doctor.Qualification}
        Age={doctor.Age}
        Speciality={doctor.Speciality}
      />
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
});