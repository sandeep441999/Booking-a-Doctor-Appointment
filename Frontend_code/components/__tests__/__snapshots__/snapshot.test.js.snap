// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li>
  <div>
    <div
      data-testid="doctor-d1"
    >
      <h3>
        Sam
      </h3>
      <address>
        <strong>
          Qualification: 
        </strong>
        MBBS
      </address>
      <address>
        <strong>
          Speciality: 
        </strong>
        Cardiology
      </address>
      <address>
        <strong>
          Age: 
        </strong>
        45
      </address>
    </div>
    <div>
      <button>
        Book an Appointment
      </button>
    </div>
  </div>
</li>
`;
